google/cloud/spanner_v1/database.py

```diff
@@ -951,20 +966,40 @@ def table(self, table_id):
         """
         return Table(table_id, self)
 
-    def list_tables(self):
+    def list_tables(self, schema="_default"):
         """List tables within the database.
 
+        :type schema: str
+        :param schema: The schema to search for tables, or None for all schemas. Use the special string "_default" to
+                       search for tables in the default schema of the database.
+
         :type: Iterable
         :returns:
             Iterable of :class:`~google.cloud.spanner_v1.table.Table`
             resources within the current database.
         """
+        if "_default" == schema:
+            schema = self.default_schema_name
+
         with self.snapshot() as snapshot:
-            if self._database_dialect == DatabaseDialect.POSTGRESQL:
-                where_clause = "WHERE TABLE_SCHEMA = 'public'"
+            if schema is None:
+                results = snapshot.execute_sql(
+                    sql=_LIST_TABLES_QUERY.format(""),
+                )
             else:
-                where_clause = "WHERE SPANNER_STATE = 'COMMITTED'"
-            results = snapshot.execute_sql(_LIST_TABLES_QUERY.format(where_clause))
+                if self._database_dialect == DatabaseDialect.POSTGRESQL:
+                    where_clause = "WHERE TABLE_SCHEMA = $1"
+                    param_name = "p1"
+                else:
+                    where_clause = (
+                        "WHERE TABLE_SCHEMA = @schema AND SPANNER_STATE = 'COMMITTED'"
```

Comment:
Dont we need table_catalog check also?

Reply:
We could add that as well, but there's no real need at the moment. We're not adding support for multiple catalogs at the moment, and are also unlikely to do so. It would also complicate things further, because the default catalog for GoogleSQL is an empty string (like for the schema), while for PostgreSQL it is the database name.